compiler/rustc_codegen_llvm/src/debuginfo/create_scope_map.rs

```diff
@@ -95,17 +95,24 @@ fn make_mir_scope<'ll, 'tcx>(
                 ty::ParamEnv::reveal_all(),
                 ty::EarlyBinder::bind(callee),
             );
-            let callee_fn_abi = cx.fn_abi_of_instance(callee, ty::List::empty());
-            cx.dbg_scope_fn(callee, callee_fn_abi, None)
+            debug_context.inlined_function_scopes.entry(callee).or_insert_with(|| {
+                let callee_fn_abi = cx.fn_abi_of_instance(callee, ty::List::empty());
+                cx.dbg_scope_fn(callee, callee_fn_abi, None)
+            })
         }
         None => unsafe {
-            llvm::LLVMRustDIBuilderCreateLexicalBlock(
-                DIB(cx),
-                parent_scope.dbg_scope,
-                file_metadata,
-                loc.line,
-                loc.col,
-            )
+            debug_context
+                .lexical_blocks
+                .entry((parent_scope.dbg_scope, loc.line, loc.col))
+                .or_insert_with(|| {
+                    llvm::LLVMRustDIBuilderCreateLexicalBlock(
+                        DIB(cx),
+                        parent_scope.dbg_scope,
+                        file_metadata,
```

Comment:
`file_metadata` is not part of the hashmap key. Should it? For instance make the key `(parent_scope.dbg_scope, log)`?

Nit: `file_metadata` and `loc` are only used in this branch, could you pull them in?

Reply:
`file_metadata` doesn't need to be included since scope (i.e., the function) + line + column uniquely identifies each lexical block, unless a function can span multiple files (which I don't believe is possible).

Reply:
A function can span multiple files if it uses macros.

Reply:
Ah good point.
Unfortunately, I can't use the `loc` as part of the key since the file in that is an `Rc`, so I'm going to use `file_metadata` instead.